[ASIM-5342] add invoke support (#316)
ESSS/alfasim-sdk: docs/source/plugins/03_plugin_structure.rst

=================

As mentioned in :ref:`quick-start-section`, the |sdk| package has some utilities to help in the process to create
a plugin file.

> First, for the creation of the plugin you can use the CLI automatic generator

Reviewer: We tend in Portuguese to be too verbose when compared to English, I suggest this which is short and to the point:

```suggestion
First, to create the plugin use :program:`new`:
```

.. code-block:: bash

   >>> alfasim-sdk new --help

.. _alfasim_sdk_cli_new_section:

.. click:: alfasim_sdk._internal.cli:new
    :prog: alfasim-sdk new
    :show-nested:

Pyinvoke tasks
--------------

> Besides the creation of the plugin struct, all other activities that you might want to perform, you will use ``pyinvoke`` tasks.

Reviewer: ```suggestion
Except for creating the plugin structure, all other activities you will use ``pyinvoke`` tasks.
```

Author: Fixed

Below there is detailed list on what they are meant to do. However, in general, the tasks that you will use the most are ``compile``,
``package`` and ``update``.

.. invoke::
   :module: alfasim_sdk.default_tasks
   :prog: invoke

You can check their implementations `here <https://github.com/ESSS/alfasim-sdk/blob/master/src/alfasim_sdk/default_tasks.py>`_

> and you can also overwrite them by just defining a function with the same name of the default task with the ``@task`` decorator.

Author: Okay, I will leave as it is.

> and you can also overwrite them by just defining a function with the same name of the default task with the ``@task`` decorator.

Reviewer 2: "read the docs" will build the docs for your PR, just add a PR comment on the same line with the link from that.

> and you can also overwrite them by just defining a function with the same name of the default task with the ``@task`` decorator.

Author: I am not sure if I understand what you are suggesting. I was talking that the link to <https://github.com/ESSS/alfasim-sdk/blob/master/src/alfasim_sdk/default_tasks.py> is broken because I am linking a file that still doesnt exist on github at the master branch.

> and you can also overwrite them by just defining a function with the same name of the default task with the ``@task`` decorator.

Author: This is a minor thing, I will resolve the conversation, because once merged the link will be valid.
